pynest/nest/tests/test_spatial/test_connect_sliced.py

```diff
@@ -0,0 +1,140 @@
+
+    def _assert_histogram(self, sources_or_targets, ref):
+        """Create a histogram of input data and assert it against reference values"""
+        hist = np.histogram(sources_or_targets, bins=self.N, range=(1, self.N+1))
+        counts, _ = hist
+        np_testing.assert_array_equal(counts, ref)
+
+    def test_connect_sliced_spatial_on_target(self):
+        """Connect sliced spatial source population"""
+        for nodes in [self.free_nodes, self.grid_nodes]:
+            nest.Connect(nodes[self.middle_node], nodes,
+                         conn_spec={'rule': 'pairwise_bernoulli',
+                                    'p': self.parameter})
+            self._assert_histogram(nest.GetConnections().target, self.reference)
+
+    def test_masked_connect_sliced_spatial_on_target(self):
+        """Masked connect sliced spatial source population"""
+        for nodes in [self.free_nodes, self.grid_nodes]:
+            nest.Connect(nodes[self.middle_node], nodes,
+                         conn_spec={'rule': 'pairwise_bernoulli',
+                                    'p': self.parameter,
+                                    'mask': self.mask})
+            self._assert_histogram(nest.GetConnections().target, self.reference_masked)
+
+    def test_connect_sliced_spatial_on_source(self):
+        """Connect sliced spatial target population"""
+        for nodes in [self.free_nodes, self.grid_nodes]:
+            nest.Connect(nodes, nodes[self.middle_node],
+                         conn_spec={'rule': 'pairwise_bernoulli',
+                                    'p': self.parameter,
+                                    'use_on_source': True})
+            self._assert_histogram(nest.GetConnections().source, self.reference)
+
+    def test_masked_connect_sliced_spatial_on_source(self):
+        """Masked connect sliced spatial target population"""
+        for nodes in [self.free_nodes, self.grid_nodes]:
+            nest.Connect(nodes, nodes[self.middle_node],
+                         conn_spec={'rule': 'pairwise_bernoulli',
+                                    'p': self.parameter,
+                                    'use_on_source': True,
+                                    'mask': self.mask})
+            self._assert_histogram(nest.GetConnections().source, self.reference_masked)
+
+    def test_sliced_spatial_inheritance(self):
+        """Sliced spatial inherits attributes"""
+        wrapped_pos = nest.spatial.free(nest.random.uniform(-0.5, 0.5), num_dimensions=2, edge_wrap=True)
+        extent_pos = nest.spatial.free(nest.random.uniform(-0.5, 0.5), extent=[5., 2.5])
+        wrapped_nodes = nest.Create('iaf_psc_alpha', positions=wrapped_pos)
+        extent_nodes = nest.Create('iaf_psc_alpha', positions=extent_pos)
+        for nodes in [self.free_nodes, self.grid_nodes, wrapped_nodes, extent_nodes]:
+            for i in range(len(nodes)):
+                nodes_sliced = nodes[i]
```

Comment:
Is the following possible?
```
for nodes_sliced in nodes:
   for attr in ['edge_wrap', 'extent']:
   ...
```

Reply:
Yes! Looks like a workaround for the rejected first slicing method. Fixed!